code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -12,50 +12,52 @@
 	if (!check_and_use_plasma_owner())
 		return
 
+	//X = xeno user, A = target atom
+	var/list/turf/target_turfs = getline2(X, A, include_from_atom = FALSE)
+	var/len = LAZYLEN(target_turfs)
+	if(len > 3)
+		target_turfs = target_turfs.Copy(1, 4)
+
 	// Get list of target mobs
 	var/list/target_mobs = list()
+	var/blocked = FALSE
 
-
-	var/list/target_turfs = list()
-	var/facing = Get_Compass_Dir(X, A)
-	var/turf/T = X.loc
-	var/turf/temp = X.loc
-
-	for(var/x in 0 to 2)
-		temp = get_step(T, facing)
-		if(!temp || temp.density || temp.opacity)
-			break
-
-		var/blocked = FALSE
-		for(var/obj/structure/S in temp)
-			if(istype(S, /obj/structure/window/framed))
-				var/obj/structure/window/framed/W = S
-				if(!W.unslashable)
-					W.deconstruct(disassembled = FALSE)
-
-			if(S.opacity)
-				blocked = TRUE
-				break
+	for(var/turf/path_turf as anything in target_turfs)
 		if(blocked)
 			break
+		//Check for walls etc and stop if we encounter one
+		if(path_turf.density)
+			break
 
-		T = temp
-		target_turfs += T
-
-	for(var/turf/target_turf in target_turfs)
-		for(var/mob/living/carbon/H in target_turf)
-			if (!isxeno_human(H) || X.can_not_harm(H))
-				continue
-
-			if(!(H in target_mobs))
-				target_mobs += H
-
-	X.visible_message(SPAN_XENODANGER("[X] slashes its claws through the area in front of it!"), SPAN_XENODANGER("You slash your claws through the area in front of you!"))
+		//Check for structures such as doors
+		for(var/atom/path_content in path_turf.contents)
+			if(istype(path_content, /obj))
+				var/obj/object = path_content
+				//If we shouldn't be able to pass through it then stop at this turf
+				if(object.density && !object.throwpass)
+					blocked = TRUE
+					break
+
+				if(istype(object, /obj/structure/window/framed))
+					var/obj/structure/window/framed/W = object
+					if(!W.unslashable)
+						W.deconstruct(disassembled = FALSE)
```

Comment:
https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STYLES.md#dont-use-abbreviations

Reply:
This is just copypasted code from the original file. I can however change it if needed.

Reply:
yeah, you should fix all the one letter vars in that proc.

Reply:
Addressed in latest commit.